sklearn/tests/test_metadata_routing.py

```diff
@@ -844,3 +847,79 @@ def test_metadata_routing_get_param_names():
     assert router._get_param_names(
         method="fit", return_alias=True, ignore_self=True
     ) == router._get_param_names(method="fit", return_alias=False, ignore_self=True)
+
+
+def test_method_generation():
+    # Test if all required request methods are generated.
+
+    # TODO: these test classes can be moved to sklearn.utils._testing once we
+    # have a better idea of what the commonly used classes are.
+    class SimpleEstimator(BaseEstimator):
+        # This class should have no set_{method}_request
+        def fit(self, X, y):
+            pass
+
+        def partial_fit(self, X, y):
+            pass
+
+        def predict(self, X):
+            pass
+
+        def predict_proba(self, X):
+            pass
+
+        def predict_log_proba(self, X):
+            pass
+
+        def decision_function(self, X):
+            pass
+
+        def score(self, X, y):
+            pass
+
+        def split(self, X, y=None):
+            pass
+
+        def transform(self, X):
+            pass
+
+        def inverse_transform(self, X):
+            pass
+
+    for method in METHODS:
+        assert not hasattr(SimpleEstimator(), f"set_{method}_request")
+
+    class SimpleEstimator(BaseEstimator):
+        # This class should have every set_{method}_request
+        def fit(self, X, y, sample_weight=None):
+            pass
+
+        def partial_fit(self, X, y, sample_weight=None):
+            pass
+
+        def predict(self, X, sample_weight=None):
+            pass
+
+        def predict_proba(self, X, sample_weight=None):
+            pass
+
+        def predict_log_proba(self, X, sample_weight=None):
+            pass
+
+        def decision_function(self, X, sample_weight=None):
+            pass
+
+        def score(self, X, y, sample_weight=None):
+            pass
+
+        def split(self, X, y=None, sample_weight=None):
+            pass
+
+        def transform(self, X, sample_weight=None):
```

Comment:
What does meta-data routing do with `y=None` in the transformers?

Reply:
`self, X, y` are hard coded to be excluded from routing.

Reply:
https://github.com/scikit-learn/scikit-learn/blob/e80be0bbf757fb9018812305c409731e73416e08/sklearn/utils/_metadata_requests.py#L1038-L1040

Reply:
Sure: https://github.com/scikit-learn/enhancement_proposals/pull/69